Renamed pip_deps to container_pip_deps

mmikitka opened the pull request:
Related to issue: https://github.com/bazelbuild/rules_docker/issues/1404

Steps to reproduce:

```
$ ./bazel version
Build label: 3.4.1
Build target: bazel-out/k8-opt/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Tue Jul 14 06:27:53 2020 (1594708073)
Build timestamp: 1594708073
Build timestamp as int: 1594708073
```

WORKSPACE
```
load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")

http_archive(
    name = "rules_python",
    url = "https://github.com/bazelbuild/rules_python/releases/download/0.0.2/rules_python-0.0.2.tar.gz",
    strip_prefix = "rules_python-0.0.2",
    sha256 = "b5668cde8bb6e3515057ef465a35ad712214962f0b3a314e551204266c7be90c",
)

load("@rules_python//python:repositories.bzl", "py_repositories")

py_repositories()

load("@rules_python//python:pip.bzl", "pip_repositories", "pip_import", "pip3_import")

pip_repositories()

pip3_import(
    name = "pip_deps",
    requirements = "//:requirements.txt",
)

load("@pip_deps//:requirements.bzl", "pip_install")

pip_install()

http_archive(
    name = "io_bazel_rules_docker",
    sha256 = "4521794f0fba2e20f3bf15846ab5e01d5332e587e9ce81629c7f96c793bb7036",
    strip_prefix = "rules_docker-0.14.4",
    urls = ["https://github.com/bazelbuild/rules_docker/releases/download/v0.14.4/rules_docker-v0.14.4.tar.gz"],
)

load(
    "@io_bazel_rules_docker//repositories:repositories.bzl",
    container_repositories = "repositories",
)
container_repositories()

load("@io_bazel_rules_docker//repositories:deps.bzl", container_deps = "deps")

container_deps()

load("@io_bazel_rules_docker//repositories:pip_repositories.bzl", container_pip_deps = "pip_deps")

container_pip_deps()

load(
    "@io_bazel_rules_docker//python:image.bzl",
    _py_image_repos = "repositories",
)

_py_image_repos()
```

BUILD
```
load("@io_bazel_rules_docker//python:image.bzl", "py_image")

py_image(
    name = "test",
    srcs = ["test.py"],
    main = "test.py",
)
```

Error message:
```
$ ./bazel build //...
ERROR: /home/mmikitka/.cache/bazel/_bazel_mmikitka/a29d44de5562ad5fee0bf3dd38d43719/external/pip_deps/requirements.bzl:19:9: Traceback (most recent call last):
	File "/home/mmikitka/.cache/bazel/_bazel_mmikitka/a29d44de5562ad5fee0bf3dd38d43719/external/io_bazel_rules_docker/container/BUILD", line 41
		py_library(<5 more arguments>)
	File "/home/mmikitka/.cache/bazel/_bazel_mmikitka/a29d44de5562ad5fee0bf3dd38d43719/external/io_bazel_rules_docker/container/BUILD", line 47, in py_library
		requirement("python-gflags")
	File "/home/mmikitka/.cache/bazel/_bazel_mmikitka/a29d44de5562ad5fee0bf3dd38d43719/external/pip_deps/requirements.bzl", line 19, in requirement
		fail(<1 more arguments>)
Could not find pip-provided dependency: 'python-gflags'
ERROR: /home/mmikitka/src/rules-docker-pip/BUILD:3:9: every rule of type _app_layer implicitly depends upon the target '@io_bazel_rules_docker//container:build_tar', but this target could not be found because of: Target '@io_bazel_rules_docker//container:build_tar' contains an error and its package is in error
ERROR: /home/mmikitka/src/rules-docker-pip/BUILD:3:9: every rule of type _app_layer implicitly depends upon the target '@io_bazel_rules_docker//container:incremental_load_template', but this target could not be found because of: Target '@io_bazel_rules_docker//container:incremental_load_template' contains an error and its package is in error
ERROR: Analysis of target '//:test' failed; build aborted: Analysis failed
INFO: Elapsed time: 64.466s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (33 packages loaded, 363 targets configured)
```

mmikitka commented:
/assign @alex1545

mmikitka commented:
@alex1545 I am seeing the following errors in buildkite, and I don't know what I did to cause them:

```

FAIL: test_py_image_complex (__main__.ImageTest)
--



```

pvcnt commented:
I was about to submit a similar PR, thank you for doing this!

smukherj1 commented:
Thanks for making this PR! Not really sure where that extra slash at the end of `/app/testdata/py_image_complex.binary.runfiles/container_pip_deps_pypi__addict_2_1_2/addict-2.1.2.dist-info/` is coming from. I'm not sure if repo named `pip_deps` mean something special to the Bazel python rules and whether this renaming will break things. @brandjon any idea if renaming the `pip_deps` repository is advisable? The repository naming conflict issue for `pip_deps` likely affects other users of `pip_deps` as well.

aaliddell commented:
Could this be named under the rules_docker 'namespace', rather than taking another name that is non-specific and still liable to clashes? i.e `rules_docker_pip_deps`. This is how it is done in other repos to prevent similar clashes and since the user doesn't typically see this name it doesn't need to be kept artificially short.

Regarding the test failures, the name `pip_deps` does not mean anything special; so using any other name is fine so long as you are consistent.

smukherj1 commented:
Ok please update the image_test.py file with the new expected name that includes the slash. I do like the idea of naming the deps with rules_docker in the name. Maybe call it `io_bazel_rules_docker_pip_deps` because `io_bazel_rules_docker` is the name of the rules_docker Bazel workspace?

mmikitka commented:
@aaliddell and @smukherj1 I renamed `container_pip_deps` to `io_bazel_rules_docker_pip_deps` and fixed the unit test error.

Anything else required to complete the PR?

smukherj1 reviewed (approved):
Thanks a lot!

smukherj1 commented:
/gcbrun

mikedanese commented on README.md:
```diff
@@ -167,9 +167,9 @@ load("@io_bazel_rules_docker//repositories:deps.bzl", container_deps = "deps")
 container_deps()
 
-load("@io_bazel_rules_docker//repositories:pip_repositories.bzl", "pip_deps")
+load("@io_bazel_rules_docker//repositories:pip_repositories.bzl", "io_bazel_rules_docker_pip_deps")
```
This snippet is using `rules_docker-0.14.4` which doesn't have your rename so copy and pasting now fails.

```
ERROR: Traceback (most recent call last):
        File "/usr/local/google/home/mikedanese/code/cloud-provider-gcp/WORKSPACE", line 70, column 89, in <toplevel>
                load("@io_bazel_rules_docker//repositories:pip_repositories.bzl", container_pip_deps = "io_bazel_rules_docker_pip_deps")
Error: file '@io_bazel_rules_docker//repositories:pip_repositories.bzl' does not contain symbol 'io_bazel_rules_docker_pip_deps'
ERROR: error loading package 'external': Package 'external' contains errors
```

mmikitka replied:
@smukherj1 and @aaliddell Tagging another release seems like a good idea, and the README.md could be updated at the same time.

smukherj1 replied:
Ack. Made a note of this and will get around to this in the next week or so hopefully.

dmattia replied:
friendly ping :)
